sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -107,10 +108,12 @@ private[execution] object HashedRelation {
         0)
     }
 
-    if (key.length == 1 && key.head.dataType == LongType) {
-      LongHashedRelation(input, key, sizeEstimate, mm)
+    if (isNullAware && !input.hasNext) {
+      new EmptyHashedRelation
+    } else if (key.length == 1 && key.head.dataType == LongType) {
+      LongHashedRelation(input, key, sizeEstimate, mm, isNullAware)
     } else {
-      UnsafeHashedRelation(input, key, sizeEstimate, mm)
+      UnsafeHashedRelation(input, key, sizeEstimate, mm, isNullAware)
```

Comment:
In the previous version of this PR: `isNullAware` used to be a flag to the hashed relation to make it not skip over null keys. 

I am not sure if `isNullAware` is still used in the `UnsafeHashedRelation` and the `LongHashedRelation` ? Apologies if I missed where it is being used still.

Reply:
for now, i am using EmptyHashedRelation and EmptyHashedRelationWithAllNullKeys to represent isOriginalInputEmpty, allNullColumnKeyExistsInOriginalInput, isNullAware should be passed into UnsafeHashedRelation and LongHashedRelation, and during building the relation, it might just return a EmptyHashedRelationWithAllNullKeys if a row with all null column key hits.